devicetypes/smartthings/ecolink-wireless-siren.src/ecolink-wireless-siren.groovy

```diff
@@ -0,0 +1,184 @@
+			}
+		}
+
+		childDeviceTiles("endpoints")
+
+		standardTile("refresh", "device.switch", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+	}
+}
+
+def installed() {
+	addChildSwitches()
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if (!childDevices) {
+		addChildSwitches()
+	}
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	off()
+	setChildSirensOff()
+}
+
+def parse(String description) {
+
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+		
+		if(result!=null) {
+			createEvent(result)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def srcEndpoint = cmd.sourceEndPoint
+	def destEnd = cmd.destinationEndPoint
+	
+	if(srcEndpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		String childDni = "${device.deviceNetworkId}-ep$srcEndpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+
+		child?.handleZWave(encapsulatedCommand)
+	}
+}
+
+def on() {
+	sendEvent(name: "switch", value: "on")
+	def cmds = []
+	cmds << basicSetCmd(0xFF, 1)
+	cmds << basicGetCmd(1)
+	delayBetween(cmds, 100)
+}
+
+def off() {
+	sendEvent(name: "switch", value: "off")
+	def cmds = []
+	cmds << basicSetCmd(0x00, 1)
+	cmds << basicGetCmd(1)
+	delayBetween(cmds, 100)
+}
+
+def setChildSirensOff() {
+	def cmds = []
+	
+	endPoints.each {
+		cmds << basicSetCmd(0x00, it)
+		cmds << basicGetCmd(it)
+	}
+	delayBetween(cmds, 50)
+}
+
+def addChildSwitches() {
+	for (i in 2..4) {
+		String childDni = "${device.deviceNetworkId}-ep$i"
+		String componentLabel =	 "$device.displayName $i"
+		String ch = "ch$i"
+		
+		addChildDevice("Z-Wave Binary Switch Endpoint Siren", childDni, device.hub.id,[completedSetup: true, label: componentLabel, isComponent: true, componentName: ch, componentLabel: componentLabel])
+	}
+}
+
+def getEndPoints() { [2, 3, 4] }
+
+def basicSetCmd(value, endPoint) {
+	multiChannelCmdEncapCmd(zwave.basicV1.basicSet(value: value), endPoint)
+}
+
+def basicGetCmd(endPoint) {
+	multiChannelCmdEncapCmd(zwave.basicV1.basicGet(), endPoint)
+}
+
+def multiChannelCmdEncapCmd(cmd, endPoint) {	
+	def cmds = []
+	cmds << zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endPoint).encapsulate(cmd).format()
+}
+
+def sendCommand(device, commands) {	 
```

Comment:
why not just have the child devices send their endpoint number instead of parsing it out of the DNI?